### PR TITLE
feat(lights): Interior and exterior lighting rework and new glass textures

### LINKED ISSUE #6108: Wing Light Enhancements
### Aircraft Version

Experimental

### Description

There are a number of inaccuracies with the current wing light. All inaccuracies will be pointed out in References with a shot or video from the IRL A320neo.

### References (optional)

Fan blades are also illuminated
engine

Stronger/Brighter light source
exterior

Engine shot
engine 2

Sources:
https://www.youtube.com/watch?v=bgwhsJeCQPQ
https://www.youtube.com/watch?v=EZkv5-eDQWs
https://www.youtube.com/watch?v=l7D_AqwDYBY

### Additional info (optional)

_No response_

### Discord Username (optional)

Aviation Spotter#9102

### PR DESCRIPTION
**Pull request by FinalLightNL**
Fixes #6108 

## Summary of Changes
My previous PR caused a couple issues looking very unrealistic in some conditions based on time of day and weather.
I have removed every single light within the cockpit and redid everything from scratch.
Including the emissive decals which has a little surprise.

List of changes :

- Removed all soft lights from all .fx files which results in proper reflections on all materials without dulling them out, this also makes textures look better imo.
- All flood types now cast ambient lights giving a ray tracing ish feel.
- Flood lights are enhanced, also now has the correct 2800K temp color according to ref data.
- Dome lights has been enhanced as well, and the first step is less bright as it should be.
- Emissive decals now have integ bulb wear below the diffusers, went of by a 7 year old NEO so didn't make it too prominent.
- The clock now powers on bus and has a blue ish background glow as per panel type, it was on @ C&D before which isn't normal behavior.
- Winglights have been changed according to references, and also cast slightly within the pod and blades (inner outside more)
- New glass textures added thanks to @MoreRightRudder for the wonderful work.
- The box shadow css related to homecockpit has been fully commented out, makes homecockpit mode useless atm (needs to be removed in another pr)
This was a necessary step to make the glass textures look as they are.
- Fault decals were yellow, changed to amber per references as well.

## Screenshots (if necessary)
NewGlass
Floods_Main1
FloodPed1
NewIntegs_1
Ambiance1
Ambiance2
Console
Dome1
Dome3
Dome2
Dome4
winglight_2png
winglight_1
ClockOnBus
Ambiance3
IntegBreakers

## Additional context
There are quite a few changes in here, but it improves the whole feel of the cockpit.
Feel free to DM me on discord or comment below when you encounter issues or see something that is off.

Discord username (if different from GitHub): FinalLight#2113

## Testing instructions
Please spawn in several different time of day, fly through clouds as well.
Play with weather presets in all conditions including fog and see if all looks fine.
check if the clock display is off when plane is in C&D and turns on when both batteries are pressed with bus power.
Check if integ lights redshift or not, this was the biggest issue in my previous PR.
They should stay between Amber Orange and yellow~ish.
Dome lights should be dim on the first position and quite bright in the second as well.

Note: May be bugs with DLSS and integrated lighting (this is considered a sim graphics issue and not related to this addon).

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/effects/A32NX_Cockpit_ConsoleMid.fx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+[LightAttributes.0]
+Type=spot
+Size=0.02
+Range=0.8
+Intensity=5
+Softness=0
+SpotInner=0
+SpotOuter=83
+Volumetric=1
+ScatDir=1
```
Comment (tracernz) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/effects/A32NX_Cockpit_ConsoleMid.fx:77]: Missing newline.

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/effects/A32NX_Cockpit_DomeLightAmbient1.fx [hunk 2]
```diff
@@ -67,11 +67,11 @@ Extrude Heading Max=0.00
...
 Softness=0.0
-SpotInner=28
-SpotOuter=57
-Volumetric=5
-ScatDir=1
\ No newline at end of file
+SpotInner=0
+SpotOuter=62
+Volumetric=0
+ScatDir=0
```
Comment (tracernz) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/effects/A32NX_Cockpit_DomeLightAmbient1.fx:77]: Missing newline.

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/effects/A32NX_Cockpit_DomeLightAmbient2.fx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+[LightAttributes.0]
+Type=spot
+Size=0
+Range=1.8
+Intensity=10
+Softness=0.0
+SpotInner=0
+SpotOuter=62
+Volumetric=0
+ScatDir=0
```
Comment (tracernz) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/effects/A32NX_Cockpit_DomeLightAmbient2.fx:77]: Missing newline.

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/effects/A32NX_Cockpit_MainPanelFlood.fx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+[LightAttributes.0]
+Type=spot
+Size=0.001
+Range=0.480
+Intensity=10.0
+Softness=0.0
+SpotInner=0
+SpotOuter=66.2
+Volumetric=0
+ScatDir=0
```
Comment (tracernz) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/effects/A32NX_Cockpit_MainPanelFlood.fx:77]: Missing newline.

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/effects/A32NX_Cockpit_MainPanelFloodCenter.fx [hunk 5]
```diff
@@ -67,11 +67,11 @@ Extrude Heading Max=0.00
...
-SpotOuter=27.0
+Size=0.001
+Range=0.47
+Intensity=75
+Softness=0
+SpotInner=16
+SpotOuter=38.3
 Volumetric=0
-ScatDir=0.0
+ScatDir=0.0
```
Comment (tracernz) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/effects/A32NX_Cockpit_MainPanelFloodCenter.fx:77]: Missing linefeed.

File: fbw-a32nx/src/systems/instruments/src/Clock/Clock.tsx [hunk 6]
```diff
@@ -18,12 +18,11 @@ export class ClockRoot extends DisplayComponent<ClockProps> {
 
         const sub = this.props.bus.getSubscriber<ClockSimvars>();
         sub.on('dcEssIsPowered').whenChanged().handle((dccEssIsPowered) => {
+            const svgElement = this.gElementRef.instance.closest('svg');
```
Comment (tracernz) [inline fbw-a32nx/src/systems/instruments/src/Clock/Clock.tsx:21]: Why not just add a ref to the element you need?

File: fbw-a32nx/src/systems/instruments/src/Common/pixels.scss [hunk 7]
```diff
@@ -8,3 +9,4 @@
   background-color: rgba(0, 0, 255, 0.035);
   box-shadow: inset 0px 0px 30px 10px rgba(0, 76, 255, 0.08);
 }
+*/
```
Comment (tracernz) [inline fbw-a32nx/src/systems/instruments/src/Common/pixels.scss:12]: Missing linefeed.

File: fbw-a32nx/src/systems/instruments/src/MsfsAvionicsCommon/pixels.scss [hunk 8]
```diff
@@ -8,3 +9,4 @@
   background-color: rgba(0, 0, 255, 0.035);
   box-shadow: inset 0px 0px 30px 10px rgba(0, 76, 255, 0.08);
 }
+*/
```
Comment (tracernz) [inline fbw-a32nx/src/systems/instruments/src/MsfsAvionicsCommon/pixels.scss:12]: Missing linefeed.

File: fbw-a32nx/src/systems/instruments/src/Common/pixels.scss [hunk 9]
```diff
@@ -1,3 +1,4 @@
+/* TO BE REMOVED TOGETHER WITH HOME COCKPIT MODE
```
Comment (2hwk) [inline fbw-a32nx/src/systems/instruments/src/Common/pixels.scss:1]:
```suggestion
/* TODO FIXME: TO BE REMOVED TOGETHER WITH HOME COCKPIT MODE
```

prefix with TODO FIXME:

File: fbw-a32nx/src/systems/instruments/src/MsfsAvionicsCommon/pixels.scss [hunk 10]
```diff
@@ -1,3 +1,4 @@
+/* TO BE REMOVED TOGETHER WITH HOME COCKPIT MODE
```
Comment (2hwk) [inline fbw-a32nx/src/systems/instruments/src/MsfsAvionicsCommon/pixels.scss:1]:
```suggestion
/*  TODO FIXME: TO BE REMOVED TOGETHER WITH HOME COCKPIT MODE
```
prefix with TODO FIXME:

**Review by 2hwk — CHANGES_REQUESTED** [review]

File: fbw-a32nx/src/systems/instruments/src/Clock/Clock.tsx [hunk 11]
```diff
@@ -12,18 +12,19 @@ interface ClockProps extends ComponentProps {
...
-                this.gElementRef.instance.classList.remove('off');
+                this.svgElementRef.instance.classList.add('powered');
+                this.svgElementRef.instance.classList.remove('unpowered');
             } else {
-                this.gElementRef.instance.classList.add('off');
-                this.gElementRef.instance.classList.remove('on');
+                this.svgElementRef.instance.classList.add('unpowered');
+                this.svgElementRef.instance.classList.remove('powered');
             }
         });
```
Comment (tracernz) [inline fbw-a32nx/src/systems/instruments/src/Clock/Clock.tsx:29]: Could be simplified if you like, up to you.
```ts
        sub.on('dcEssIsPowered').whenChanged().handle((dccEssIsPowered) => {
            this.svgElementRef.instance.classList.toggle('powered', dccEssIsPowered);
            this.svgElementRef.instance.classList.toggle('unpowered', !dccEssIsPowered);
        });
```

**Comment by 2hwk** [comment]
Lint is failing (Line 15 Clock.tsx)

**Review by 2hwk — APPROVED** [review]

**Comment by 2hwk** [comment]
I've run through the testing instructions once but I feel this needs also more testing with wider hardware and software configurations at various resolutions. Definitely candidate for EXP for wider testing and feedback.

**Comment by 2hwk** [comment]
QA Report

Testing:
- [x] Please spawn in several different time of day, fly through clouds as well.
- [x] Play with weather presets in all conditions including fog and see if all looks fine.
- [x] Check if the clock display is off when plane is in C&D and turns on when both batteries are pressed with bus power.
- [x] Check if integ lights redshift or not, this was the biggest issue in my previous PR. They should stay between Amber Orange and yellow~ish.
- [x] Dome lights should be dim on the first position and quite bright in the second as well.

Note:
* This run was done on proton experimental with DXVK as such, testing is possibly only valid for functionality and not visuals, as there are quite substantial differences in visuals between DXVK and DX11 on windows.

Settings were custom/Ultra at 1440p (No 10-bit or HDR). Unknown how visuals would be affected at lower resolutions (1080p). 

Negatives:
- May need adjustment from previously set user lighting presets, generally would need to be set darker. This should be communicated.

Result:
QA Pass, but see note

Conclusions:
LGTM but push to experimental for wider public testing.

**Comment by Ker1010** [comment]
hello I'm testing light new at exp mode 24d7452 version then when look at external found a strange light above the cockpit
image

turn on wing light
image

turn off wing light
image


https://user-images.githubusercontent.com/42411469/230036874-2dee8318-c62c-420f-a221-b387522e5639.mp4

Great work!

**Comment by FinalLightNL** [comment]
if anyone else spots floating tiny orbs let me know, i fixed the one Ker reported and found 2 more which i fixed.

**Comment by alexr4339** [comment]
Quality Assurance Tester Report

Discord          : Alex R#6663
Object of testing: #PR 7947
Tier of Testing  : 2
Date             : 06/04/2023

Testing Process: 
Testing process as described in testing instruction. Simulator is running on DX11.

Negatives:       
No negatives found. The flood light (the flood light coming from the bottom of the glareshield which illuminates the area around the displays) at full brightness during night seemed a bit bright to me. That may be due to my system. Additionally I dont have a personal reference of how bright the flood light is at night and full brightness. As a result this statement should be taken as a personal preference only.

Testing Results: 
Passed

Conclusions: 
This PR improves the lighting greatly and improves the feeling of the lighting.

**Comment by Watsi01** [comment]
Again we have to discuss the glass texturing (imho). It was the same about 1 year ago or so - and got reverted back because of user feedback.
Maybe the reflections are fine. But if changes in the glass textures result in strange coloring of the displays (again > imho), I would assume users wouldn't appreciate it. 

See this cyan coloring on top of the FO ND?
glass

Have the same on top of the CPT PFD. Regardless which angle/position I look on it.

If this is not done yet, forget what I wrote and I'll wait. 

I just think we have to be cautious to have the coloring pretty accurate, regardless the angle or outside light effecting it. This may be more unrealistic according to the real look in a real cockpit, but this way it is kind of disturbing in the sim (imho :-) )

**Comment by MoreRightRudder** [comment]
> 

Hi the display coloring is carefully made and are accurate to displays like these. The display fades over time and the pixels don't output the same as they used to. This leads to a color degradation which I have depicted here. Feel free to make transparent the glass texture in your file using paint.net or delete the file from your texture folder. This will get rid of the glass texture. 

Additionally I made these textures more than one year ago. Now they are finally merged. I do not plan to address any more a320 textures until a new model is worked on, thank you.

**Comment by Watsi01** [comment]
> > 
> 
> Hi the display coloring is carefully made and are accurate to displays like these. The display fades over time and the pixels don't output the same as they used to. This leads to a color degradation which I have depicted here. Feel free to make transparent the glass texture in your file using paint.net or delete the file from your texture folder. This will get rid of the glass texture.
> 
> Additionally I made these textures more than one year ago. Now they are finally merged. I do not plan to address any more a320 textures until a new model is worked on, thank you.

Hi Mike,
I'm really sorry, if I hurt your feelings. That was not my intention. Nore was it to critisize your excelent work. My intention was to discuss, how real it should get / can get in a sim. And if really everything that may look, sound or feel like the real thing is appropriate to port it to a simulation. Are there limits? Or maybe not?
I had a similar discussion some while ago regarding sounds. Some things can sound horrible (or disturbing) in a real cockpit. Is it indispensable to port them into a sim and let the sim-pilot hear that disturbing sound all the time and in real loudness? Or should one decide to make a compromise between reality and ear candy?

**Pull request merged**